src/backend/base/langflow/graph/graph/utils.py

```diff
@@ -431,13 +431,14 @@ def should_continue(yielded_counts: dict[str, int], max_iterations: int | None)
 
 
 def find_cycle_vertices(edges):
-    # Create a directed graph from the edges
     graph = nx.DiGraph(edges)
 
-    # Find all simple cycles in the graph
-    cycles = list(nx.simple_cycles(graph))
+    # Initialize a set to collect vertices part of any cycle
+    cycle_vertices = set()
 
-    # Flatten the list of cycles and remove duplicates
-    cycle_vertices = {vertex for cycle in cycles for vertex in cycle}
+    # Utilize the strong component feature in NetworkX to find cycles
+    for component in nx.strongly_connected_components(graph):
+        if len(component) > 1 or graph.has_edge(tuple(component)[0], tuple(component)[0]):
```

Comment:
`next()` isn't correct because it will increment the iterable, and we need to access the same element twice. I will disable this lint rule

Reply:
Sounds good